akka-actor-tests/src/test/scala/akka/pattern/AskSpec.scala

```diff
@@ -111,6 +111,45 @@ class AskSpec extends AkkaSpec {
       Await.result(identityFuture, 5 seconds) should be(echo)
     }
 
+    "work when reply uses actor selection" in {
+      implicit val timeout = Timeout(5 seconds)
+      import system.dispatcher
+      val echo = system.actorOf(Props(new Actor { def receive = { case x ⇒ context.actorSelection(sender().path) ! x } }), "select-echo2")
+      val r = echo ? "hi"
+
+      Await.result(r, 5 seconds) should be("hi")
+    }
+
+    "work when reply uses actor selection wildcard pattern" in {
+      implicit val timeout = Timeout(5 seconds)
+      import system.dispatcher
+      val echo = system.actorOf(Props(new Actor {
+        def receive = {
+          case x ⇒
+            val path = sender().path
+            context.actorSelection(path / ".." / "*") ! x
```

Comment:
This will complete all outstanding “ask” operations in the system (that had their paths registered), so I don’t think this is a valid test.